### PR TITLE
Allow translating rich-text attributes values

### PR DESCRIPTION
**Pull request by dexon44**
I want to merge this change because it allows to translate product and page attributes.

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3738 Allow translating product and page attributes](https://app.clickup.com/t/2549495/SALEOR-3738)

File: saleor/graphql/checkout/mutations.py [hunk 1]
```diff
@@ -640,6 +640,11 @@ def perform_mutation(
                 info, checkout_id or token, only_type=Checkout, field="checkout_id"
             )
 
+        # Raise error when trying to attach a user to a checkout
+        # that is already owned by another user.
+        if checkout.user:
+            raise PermissionDenied()
+
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations.py:647]: These changes are unrelated to this PR.

File: saleor/graphql/checkout/tests/test_checkout.py [hunk 2]
```diff
@@ -1506,6 +1506,43 @@ def test_checkout_customer_attach(
     assert checkout.email == customer_user.email
 
 
+def test_checkout_customer_attach_user_to_checkout_with_user(
```
Comment (maarcingebala) [inline saleor/graphql/checkout/tests/test_checkout.py:1509]: Same as above.

File: saleor/graphql/translations/tests/test_translations.py [hunk 3]
```diff
@@ -2308,3 +2308,189 @@ def test_product_and_attribute_translation(user_api_client, product, channel_USD
...
+    #                         name
+    #                         richText
+    #                         }
+    #                     }
+    #                 }
+    #             }
+    #         }
+    #     }
+    # }
+    # """
```
Comment (maarcingebala) [inline saleor/graphql/translations/tests/test_translations.py:2367]: 🚓

File: saleor/graphql/translations/tests/test_translations.py [hunk 4]
```diff
@@ -2308,3 +2308,189 @@ def test_product_and_attribute_translation(user_api_client, product, channel_USD
...
+    )
+
+    query = """
+    query translation(
+        $kind: TranslatableKinds!, $id: ID!, $languageCode: LanguageCodeEnum!
+    ){
+        translation(kind: $kind, id: $id){
+            __typename
+            ...on ProductVariantTranslatableContent{
+                        name
```
Comment (maarcingebala) [inline saleor/graphql/translations/tests/test_translations.py:2411]: Please fix the formatting (indent) here and in other tests.

File: saleor/graphql/translations/tests/test_translations.py [hunk 5]
```diff
@@ -2308,3 +2308,189 @@ def test_product_and_attribute_translation(user_api_client, product, channel_USD
...
+    data = get_graphql_content(response)["data"]
+
+    attribute_value_response = data["translation"]["attributeValues"][0]
+    assert attribute_value_response["name"] == attribute_value.name
+    assert attribute_value_response["richText"] == dummy_editorjs(
+        "Rich text attribute content.", json_format=True
+    )
+    assert attribute_value_response["translation"]["richText"] == dummy_editorjs(
+        "Test_dummy_data", json_format=True
+    )
```
Comment (maarcingebala) [inline saleor/graphql/translations/tests/test_translations.py:2496]: Not a big issue, but it would be easier to read these tests if we used actual translations of the original strings. Here we have two random English strings and it's harder to guess what output to expect, just by reading the code.

File: saleor/attribute/__init__.py [hunk 6]
```diff
@@ -38,6 +38,11 @@ class AttributeInputType:
         NUMERIC,
     ]
 
+    # list of the translatable attributes
```
Comment (maarcingebala) [inline saleor/attribute/__init__.py:41]: Dropdown and mulitselects are also translatable, but we don't include them here. It would be good to mention that:

```suggestion
    # list of the translatable attributes, excluding attributes with choices
```

File: saleor/graphql/translations/tests/test_translations.py [hunk 7]
```diff
@@ -2308,3 +2308,189 @@ def test_product_and_attribute_translation(user_api_client, product, channel_USD
...
+    query = """
+    query translation(
+        $kind: TranslatableKinds!, $id: ID!, $languageCode: LanguageCodeEnum!
+    ){
+        translation(kind: $kind, id: $id){
+            __typename
+            ...on ProductTranslatableContent{
+                name
+                attributeValues{
+                name
```
Comment (tomaszszymanski129) [inline saleor/graphql/translations/tests/test_translations.py]: you're already asking for `name` in line 2334

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/translations/tests/test_translations.py [hunk 8]
```diff
@@ -2308,3 +2308,189 @@ def test_product_and_attribute_translation(user_api_client, product, channel_USD
...
+    product_id = graphene.Node.to_global_id(
+        "Product", product_with_rich_text_attribute[0].id
+    )
+
+    query = """
+    query translation(
+        $kind: TranslatableKinds!, $id: ID!, $languageCode: LanguageCodeEnum!
+    ){
+        translation(kind: $kind, id: $id){
+            __typename
```
Comment (tomaszszymanski129) [inline saleor/graphql/translations/tests/test_translations.py]: don't ask for properties you don't test

File: saleor/graphql/translations/tests/test_translations.py [hunk 9]
```diff
@@ -2308,3 +2308,189 @@ def test_product_and_attribute_translation(user_api_client, product, channel_USD
...
+):
+    rich_text = dummy_editorjs("Test_dummy_data")
+    variant_attr = page_type_with_rich_text_attribute.page_attributes.first()
+    attribute_value = variant_attr.values.first()
+    attribute_value.translations.create(language_code="pl", rich_text=rich_text)
+
+    page_id = graphene.Node.to_global_id("Page", page_with_rich_text_attribute.id)
+
+    query = """
+    query translation(
```
Comment (tomaszszymanski129) [inline saleor/graphql/translations/tests/test_translations.py]: the query is the same as in other tests, no? extract it to constant somewhere before these tests, don't repeat it like that in each unit.

File: saleor/graphql/translations/types.py [hunk 10]
```diff
@@ -138,6 +144,16 @@ class Meta:
...
 
+    @staticmethod
+    def resolve_attribute_values(root: product_models.ProductVariant, _info):
+        translatable_values = []
+        for assignment in root.attributes.all():
+            attr = assignment.attribute
+            if attr.input_type in AttributeInputType.TRANSLATABLE_ATTRIBUTES:
+                value = assignment.values.first()
+                translatable_values.append(value)
+        return translatable_values
```
Comment (tomaszszymanski129) [inline saleor/graphql/translations/types.py:155]: Maybe shorter version, but im not sure if it fits the line length limit
```
return [
    assignment.values.first()
    if assignment.attribute.input_type in AttributeInputType.TRANSLATABLE_ATTRIBUTES
    for assignment in root.attributes.all()
]
```

File: saleor/graphql/translations/tests/test_translations.py (hunk 7 above)
Comment (tomaszszymanski129) [reply saleor/graphql/translations/tests/test_translations.py]: Oh, this is nested. Missed it due to formatting :/

File: saleor/graphql/translations/tests/test_translations.py [hunk 11]
```diff
@@ -2308,3 +2308,156 @@ def test_product_and_attribute_translation(user_api_client, product, channel_USD
...
+    rich_text = dummy_editorjs("Test_dummy_data")
+    variant_attr = page_type_with_rich_text_attribute.page_attributes.first()
+    attribute_value = variant_attr.values.first()
+    attribute_value.translations.create(language_code="pl", rich_text=rich_text)
+
+    page_id = graphene.Node.to_global_id("Page", page_with_rich_text_attribute.id)
+
+    query = """
+        query translation(
+            $   kind: TranslatableKinds!, $id: ID!, $languageCode: LanguageCodeEnum!
```
Comment (tomaszszymanski129) [inline saleor/graphql/translations/tests/test_translations.py:2432]: `$   kind` 🚓 👮🏻 👮🏻‍♂️ 👮🏻‍♀️ 🚔

**Review by IKarbowiak — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]
